### PR TITLE
fix(protocol): can use custom content when Dubbo encode

### PR DESCRIPTION
**Pull request by Sharember**
### Issues associated with this PR

[1304](https://github.com/mosn/mosn/issues/1304)

### Solutions
also set payload when invoke DubboFrame.SetData, and make old raw nil

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/protocol/xprotocol/dubbo/encoder.go [hunk 1]
```diff
@@ -39,10 +39,14 @@ func encodeFrame(ctx context.Context, frame *Frame) (types.IoBuffer, error) {
 		// 1.1 replace requestId
 		binary.BigEndian.PutUint64(frame.rawData[IdIdx:], frame.Id)
 
-		// hack: increase the buffer count to avoid premature recycle
-		frame.data.Count(1)
-		return frame.data, nil
+		// 1.2 check if content changed
```
[the remark this answers is not in the document]
Comment (Sharember) [reply pkg/protocol/xprotocol/dubbo/encoder.go:42]: when receive request, mosn will invoke decodeFrame first and set frame.rawData

[the remark this answers is not in the document]
Comment (champly) [reply pkg/protocol/xprotocol/dubbo/encoder.go:42]: Conversion from HTTP to dubbo situation? Maybe need set `frame.rawData`.

File: pkg/protocol/xprotocol/dubbo/command.go [hunk 2]
```diff
@@ -81,7 +83,12 @@ func (r *Frame) GetData() types.IoBuffer {
 }
 
 func (r *Frame) SetData(data types.IoBuffer) {
-	r.content = data
+	if r.content != data {
+		r.content = data
+		r.payload = data.Bytes()
+
+		r.ContentChanged = true
```
[the remark this answers is not in the document]
Comment (Sharember) [reply pkg/protocol/xprotocol/dubbo/command.go:90]: this is a problem... in fact, bolt and boltv2 have the same problem.

[the remark this answers is not in the document]
Comment (cch123) [reply pkg/protocol/xprotocol/dubbo/command.go:90]: in Chinese:
DataLen 在这里也应该修改的吧

**Comment by nejisama** [comment]
please fix the conflict

**Comment by nejisama** [comment]
we close this pr. if you still want this pr merged, fix the conflict and make a new one. thanks @Sharember

**Pull request closed without merging**